homework/src/index.js

```diff
@@ -1,47 +1,91 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'html') {
-        elem.innerHTML = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
-
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-
-  window.onload = () => main(HYF_REPOS_URL);
-}
+    function fetchJSON(url, cb) {
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', url);
+        xhr.responseType = 'json';
+        xhr.onload = () => {
+            if (xhr.status < 400) {
+                cb(null, xhr.response);
+            } else {
+                cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+            }
+        };
+        xhr.onerror = () => cb(new Error('Network request failed'));
+        xhr.send();
+    }
+
+    function createAndAppend(name, parent, options = {}) {
+        const elem = document.createElement(name);
+        parent.appendChild(elem);
+        Object.keys(options).forEach((key) => {
+            const value = options[key];
+            if (key === 'text') {
+                elem.innerHTML = value;
+            } else {
+                elem.setAttribute(key, value);
+            }
+        });
+        return elem;
+    }
+
+    function main(url) {
```

Comment:
If you look at the HTML your application generates for the `table` in the Chrome inspector you can see that it has issues:

```html
<table class="table">
  <tbody>
    <tr></tr>
    <td class="label">Repository: </td><a href="https://github.com/HackYourFuture/DataStructures" target="_blank">DataStructures</a><tr>Description:
      Algorithms and Data structures course</tr>
    <tr class="label">Forks: 12</tr>
    <tr class="label">Updated: 16/04/2018, 14:43:20</tr>
    <td></td>
  </tbody>
</table>
``` 

I'll add comments at the detail level where this goes wrong.